[masonry] Put align-tracks & justify-tracks behind separate preference

nt1m opened the pull request:
#### a40e16fe3a7e5754a7db9241628211a7f5589704
<pre>
[masonry] Put align-tracks &amp; justify-tracks behind separate preference
<a href="https://bugs.webkit.org/show_bug.cgi?id=257990">https://bugs.webkit.org/show_bug.cgi?id=257990</a>
rdar://110674890

Reviewed by Alan Baradlay and Myles C. Maxfield.

There is currently no layout support for those CSS properties, only parsing, so let&apos;s gate it behind it a separate pref so feature detection is correct.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
(GenerateCSSPropertyNames):

Canonical link: <a href="https://commits.webkit.org/265126@main">https://commits.webkit.org/265126@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4e74dea45f4d308f443a147d3b65272ca742fbcf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9925 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10172 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10422 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11577 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9622 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9933 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12158 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10124 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12568 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10079 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10878 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/8410 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11960 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8185 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9001 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16342 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/8413 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9283 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9150 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12424 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/9425 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9651 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7842 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10068 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8810 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/2574 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2375 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13036 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/10338 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9425 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2560 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4009,6 +4009,20 @@ MasonryEnabled:
+  type: bool
+  status: testable
+  category: css
+  humanReadableName: "CSS align-tracks & justify-tracks properties"
```
This should probably mention Masonry

stwrt commented on Source/WebCore/css/process-css-properties.py:
```diff
@@ -2761,7 +2761,7 @@ def _generate_css_property_settings_hasher(self, *, to):
         to.write(f"void add(Hasher& hasher, const CSSPropertySettings& settings)")
         to.write(f"{{")
         with to.indent():
-            to.write(f"unsigned bits = {first}")
```
Why are we updating this?

nt1m replied:
I was trying to fix the build failure on non-mac platforms, but it seems unsuccessful

stwrt replied:
Never looked at this code before, so really don't have much insight into why this is needed.

nt1m replied:
My latest change seemed to have fixed the build failure

stwrt reviewed (commented):
Small changes

stwrt commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -4009,6 +4009,20 @@ MasonryEnabled:
+  status: testable
+  category: css
+  humanReadableName: "CSS align-tracks & justify-tracks properties"
+  humanReadableDescription: "Enable CSS align-tracks & justify-tracks properties"
```
Also, it would be good to mention that we should have masonry flag enabled too.

nt1m replied:
done, thanks for reminding me, I was too focused on fixing the build failure

stwrt reviewed (approved):
LGTM now. Can't really comment on the `process-css-property.py` changes

nt1m committed:
[masonry] Put align-tracks & justify-tracks behind separate preference
https://bugs.webkit.org/show_bug.cgi?id=257990
rdar://110674890

Reviewed by Alan Baradlay and Myles C. Maxfield.

There is currently no layout support for those CSS properties, only parsing, so let's gate it behind it a separate pref so feature detection is correct.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:
(GenerateCSSPropertyNames):

Canonical link: https://commits.webkit.org/265126@main
